eng/packages/http-client-csharp/generator/Azure.Generator/src/Primitives/NewAzureProjectScaffolding.cs

```diff
@@ -51,25 +51,37 @@ protected override string GetSourceProjectFileContent()
                 TraverseInput(client, ref hasOperation, ref hasLongRunningOperation);
             }
 
+            var sharedFiles = new List<string>(_additionalSharedFiles);
+
             if (hasOperation)
             {
-                foreach (var file in _operationSharedFiles)
-                {
-                    builder.CompileIncludes.Add(new CSharpProjectWriter.CSProjCompileInclude(GetCompileInclude(file, pathSegmentCount), SharedSourceLinkBase));
-                }
+                sharedFiles.AddRange(_operationSharedFiles);
             }
 
             if (hasLongRunningOperation)
             {
-                foreach (var file in _lroSharedFiles)
-                {
-                    builder.CompileIncludes.Add(new CSharpProjectWriter.CSProjCompileInclude(GetCompileInclude(file, pathSegmentCount), SharedSourceLinkBase));
-                }
+                sharedFiles.AddRange(_lroSharedFiles);
+            }
+
+            foreach (var file in sharedFiles.Distinct().OrderBy(filename => filename))
+            {
+                builder.CompileIncludes.Add(new CSharpProjectWriter.CSProjCompileInclude(GetCompileInclude(file, pathSegmentCount), SharedSourceLinkBase));
             }
 
             return builder.Write();
         }
 
+        private readonly IList<string> _additionalSharedFiles = new List<string>();
+
+        /// <summary>
+        /// Add more shared files to the project scaffolding.
+        /// </summary>
+        /// <param name="filename"></param>
+        protected void AddSharedFiles(string filename)
```

Comment:
Wonder if this should be a concept in the NewProjectScaffolding - CompileIncludes. Currently it is only in the CSharpProjectWriter.